ng-dev/pr/merge/task.ts

```diff
@@ -162,11 +154,7 @@ export class PullRequestMergeTask {
       }
       throw e;
     } finally {
-      // Always try to restore the branch if possible. We don't want to leave
-      // the repository in a different state than before.
-      if (previousBranchOrRevision !== null) {
-        this.git.runGraceful(['checkout', '-f', previousBranchOrRevision]);
-      }
+      await strategy.cleanup(pullRequest, previousBranchOrRevision);
```

Comment:
Everything looks good, except that I still believe the previous branch should be checked out here explicitly and not as part of the strategy cleanup. The strategy is only concerned with cleaning up its artifacts. Could you please change that?